tests/test.rs

```diff
@@ -177,19 +179,17 @@ impl Test {
 }
 
 impl Test {
-  pub(crate) fn run(self) -> Output {
+  pub(crate) fn run(mut self) -> Output {
```

Comment:
Instead of breaking this function up, how about adding the pid to the output struct? That way, callers can all still use run, and if they care about the pid, they can get it from the output.

Reply:
Yeah, this makes much more sense.

Done.